crates/fluvio-cli/src/client/produce/mod.rs

```diff
@@ -211,6 +213,35 @@ mod cmd {
             self,
             _out: Arc<O>,
             fluvio: &Fluvio,
+        ) -> Result<()> {
+            self.process_client_impl(_out, fluvio).await.map_err(|err| {
+                error!(%err, "Error producing messages",);
+
+                let err = if let Some(producer_error) =
+                    err.downcast_ref::<ProducerError>()
+                {
+                    match producer_error {
+                        ProducerError::RecordTooLarge ( size ) => {
+                            CliError::CollectedError(format!("Record too large. Try increasing the batch size ({} bytes) or reducing the record size enabling a compression algorithm.", size)).into()
```

Comment:
instead of doing here.  Why not just add custom msg for `Producer::RecordTooLarge`?